iceberg/patched-iceberg-core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -1,420 +0,0 @@
-  }
-
-  protected Table newTable(TableOperations ops, String tableName) {
-    return new BaseTable(ops, tableName);
-  }
-
-  @Override
-  public String name() {
-    return name;
-  }
-
-  @Override
-  public String location() {
-    return location;
-  }
-
-  @Override
-  public Map<String, String> properties() {
-    return properties;
-  }
-
-  @Override
-  public Schema schema() {
-    if (lazySchema == null) {
-      synchronized (this) {
-        if (lazySchema == null && lazyTable == null) {
-          // prefer parsing JSON as opposed to loading the metadata
-          this.lazySchema = SchemaParser.fromJson(schemaAsJson);
-        } else if (lazySchema == null) {
-          this.lazySchema = lazyTable.schema();
-        }
-      }
-    }
-
-    return lazySchema;
-  }
-
-  @Override
-  public Map<Integer, Schema> schemas() {
-    return lazyTable().schemas();
-  }
-
-  @Override
-  public PartitionSpec spec() {
-    return specs().get(defaultSpecId);
-  }
-
-  @Override
-  public Map<Integer, PartitionSpec> specs() {
-    if (lazySpecs == null) {
-      synchronized (this) {
-        if (lazySpecs == null && lazyTable == null) {
-          // prefer parsing JSON as opposed to loading the metadata
-          Map<Integer, PartitionSpec> specs = Maps.newHashMapWithExpectedSize(specAsJsonMap.size());
-          specAsJsonMap.forEach(
-              (specId, specAsJson) -> {
-                specs.put(specId, PartitionSpecParser.fromJson(schema(), specAsJson));
-              });
-          this.lazySpecs = specs;
-        } else if (lazySpecs == null) {
-          this.lazySpecs = lazyTable.specs();
-        }
-      }
-    }
-
-    return lazySpecs;
-  }
-
-  @Override
-  public SortOrder sortOrder() {
-    if (lazySortOrder == null) {
-      synchronized (this) {
-        if (lazySortOrder == null && lazyTable == null) {
-          // prefer parsing JSON as opposed to loading the metadata
-          this.lazySortOrder = SortOrderParser.fromJson(schema(), sortOrderAsJson);
-        } else if (lazySortOrder == null) {
-          this.lazySortOrder = lazyTable.sortOrder();
-        }
-      }
-    }
-
-    return lazySortOrder;
-  }
-
-  @Override
-  public Map<Integer, SortOrder> sortOrders() {
-    return lazyTable().sortOrders();
-  }
-
-  @Override
-  public FileIO io() {
-    return io;
-  }
-
-  @Override
-  public EncryptionManager encryption() {
-    return encryption;
-  }
-
-  @Override
-  public LocationProvider locationProvider() {
-    return locationProvider;
-  }
-
-  @Override
-  public List<StatisticsFile> statisticsFiles() {
-    return lazyTable().statisticsFiles();
-  }
-
-  @Override
-  public Map<String, SnapshotRef> refs() {
-    return refs;
-  }
-
-  @Override
-  public void refresh() {
-    throw new UnsupportedOperationException(errorMsg("refresh"));
-  }
-
-  @Override
-  public TableScan newScan() {
-    return lazyTable().newScan();
-  }
-
-  public IncrementalAppendScan newIncrementalAppendScan() {
```

Comment:
Comparing the diff between the patched-iceberg-core-SerializableTable in hive and org.apache.iceberg.SerializableTable in iceberg .
![image](https://github.com/apache/hive/assets/26441677/89a7c068-3126-4717-8883-ee10b71ddb19)

Reply:
Can we reconfirm this and restore SerializableTable if needed?
Thanks!

Reply:
it's used in IcebergInputFormat
````
  private static IncrementalAppendScan createIncrementalAppendScan(Table table, Configuration conf) {
    long fromSnapshot = conf.getLong(InputFormatConfig.SNAPSHOT_ID_INTERVAL_FROM, -1);
    return table.newIncrementalAppendScan().fromSnapshotExclusive(fromSnapshot);
  }
````
we should have it in iceberg/patched-iceberg-core/pom.xml

Reply:
Fixed.